### PR TITLE
Add webpack config options

### PR DESCRIPTION
**Pull request by jimmynotjim**
This is a bit of an experiment based off our discussion yesterday. Let me know what you think. We may want to do some validation on the CLI side for the string that's passed to `entry`.

### CODE REVIEW

**Comment by ductiletoaster** [comment]
I like this but I want to explore what other options before we commit to this. For instance how will this work when we give them a entry point path? 

Also could using individual options help us when we want to do defaults.

For instance
```
--mode="development" maybe the default is always production or reverse
```
versus the way we have it it here makes that a bit harder because the position of those values can change if they are left out.

**Comment by jimmynotjim** [comment]
The main issue I've come across is we can't map multiple values to an option like we had sketched out yesterday, the item is ignored entirely.

```
--name client ./src/client/index.tsx 
```

results in

```
{
  name: "client"
}
```

so I think we're going to have to make the file path the last item in the entry string.

The only other option I can see is if all the flags take an array of strings but then we have to make sure everyone maps them in the same order. Ex

```
--mode production,production --name client,server --target web,node --file-path ./src/client/index.tsx,./src/server/index.tsx
```

If you were to switch any of them it would all fall apart where it's a bit easier to tell if you have the right ones mapped together as one string

File: packages/pixeloven-webpack/config/src/types.ts [hunk 1]
```diff
@@ -13,6 +13,17 @@ export interface Config {
     };
 }
 
+export interface CompilerConfig {
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack/config/src/types.ts]: @ductiletoaster let me know if you have a better name here. It's getting hard to follow what is what because there's multiple objects with similar names/structure.

File: packages/pixeloven/cli-addon-webpack/src/types.ts [hunk 2]
```diff
@@ -17,13 +17,18 @@ export interface WebpackExtensionOptions {
 
 /* Note: keep this enum up to date matching the options interface above for runtime argument checking */
 export enum WebpackExecutionOptionTypes {
+    "client" = "client",
+    "development" = "development",
+    "entry" = "entry",
```
Comment (jimmynotjim) [inline packages/pixeloven/cli-addon-webpack/src/types.ts]: Whoops, need to remove this.

File: packages/pixeloven-webpack/config/src/types.ts (hunk 1 above)
Comment (ductiletoaster) [reply packages/pixeloven-webpack/config/src/types.ts]: I don't have a good suggestion right now but I totally agree it is hard to follow.

**Comment by ductiletoaster** [comment]
In apps/pixeloven/site I ran 
```
yarn build:development
```
which does not include the stats option and I still got

**Comment by ductiletoaster** [comment]
Awe! Sorry. I pushed a fix to something I had done wrong in the webpack config. And I meant to cancel a comment I had placed but isntead closed the PR. :)

File: packages/pixeloven-webpack/compiler/src/index.ts [hunk 3]
```diff
@@ -24,27 +25,27 @@ const defaultCompilerOptions: Config = {
  * @param options
  */
 function getCompiler(options: Partial<Config> = {}) {
+    const { compilers } = options;
+    delete options.compilers;
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack/compiler/src/index.ts]: @ductiletoaster meant to ask you if this is the best way to avoid passing the compilers on to `getConfig` via the `...config` spread

File: packages/pixeloven/cli-addon-webpack/src/commands/webpack.ts [hunk 4]
```diff
@@ -33,19 +52,82 @@ export default {
...
+                                entry: options[1],
+                                target: Target[options[0]],
+                            };
+                        } else if (options[0].includes(".")) {
+                            return {
+                                entry: options[0],
+                            };
+                        } else {
+                            return {
+                                target: Target[options[0]],
```
Comment (jimmynotjim) [inline packages/pixeloven/cli-addon-webpack/src/commands/webpack.ts]: Should probably check that the Target key is supported and throw an error here if it's not.
